### ISSUE TITLE
[Feature Request] Option for the 4K requests

### ISSUE THREAD

**Issue by VP-EN** [kiranshila/Doplarr#22]
[Feature Request] Option for the 4K requests

For the sake of explanation, when I refer to "requesting in 4K" or "4K", I refer to the 4K instances and profiles of Sonarr/Radarr enabled in Overseerr. When I refer to "requesting in 1080p" or "1080p", I refer to the regular/standard instances and profiles of Sonarr/Radarr enabled in Overseerr:
https://i.imgur.com/IGTVQ36.png

**Is your feature request related to a problem? Please describe.**
Sometimes when users request items in 4K, they don't check if they're available already in 1080p in Plex. If they request an item in 4K, that's not already present in 1080p, it can sometimes cause weird issues with Overseerr.. Like Overseerr refusing to request the item in the 1080p associated profiles.

**Describe the solution you'd like**
Therefore, it would be awesome with an option in Doplarr, that does a sort of check, if a user presses "Request 4K"... Like so:
- User uses `/request movie {term}`
- User selects the movie and presses `Request 4K`, without checking if it's already on Plex in 1080p
- Doplarr checks Overseerr if the movie is already available on Plex
- If movie is NOT available on Plex, it'll request both in 1080p and in 4K
- If movie IS present on Plex, it'll just send the regular request in 4K regardless

This should also apply to series' ofc:
- User uses `/request series {term}`
- User selects the series and presses `Request 4K`, without checking if it's already on Plex in 1080p
- Doplarr checks Overseerr if the series is already available on Plex
- If series is NOT available on Plex, it'll request both in 1080p and in 4K
- If series IS present on Plex, it'll just send the regular request in 4K regardless

**Additional context**
I hope I make sense. This option would be awesome. Especially to avoid problems in Overseerr - but also because it's nice to have the movie available in both instances of Radarr/Sonarr.

**Comment by kiranshila** [comment]
Thinking more about this, would it make sense to have a `Request Missing` button? It seems counterintuitive to me to have the `Request 4K` button request both. So then with this third option you can just have it request whichever quality profile you don't have.

**Comment by VP-EN** [comment]
Hmm.. Might be. 
The reason why I'm proposing this, is because I had an issue in Overseerr, where it mistakingly thought an item was available in both 1080p and 4K, while it actually was only present in 4K.. Trying to request it in 1080p was not possible, and I had to manually add the item/movie through the default Radarr instance.

So this was a proposal on how to avoid this, by letting it request on both profiles, if the item was not already present on Plex in 1080p.

The reason I wanted it to be bound to the Request 4K button was just for simplicity. The less buttons the better, atleast for my casual discord users 😄 . I **know** it is counterintuitive, but I'm thinking this feature should be optional. 

I'm a but worried that having a third `Request Missing` button would just confuse my users, and they wouldn't pick it.

Maybe it could function like this - with a optional `Request Both` button/option:
- This "option"/"feature" is enabled
- User uses `/request movie {term}`
- User selects the movie and presses `Request 4K`, without checking if it's already on Plex in 1080p
- Doplarr checks with Overseerr if the movie is already available on Plex
- If movie is NOT available on Plex, a `Request Both` box will appear, with a message that tells the user, that the item is missing in Plex, and to request it on both profiles (using said `Request Both` button).
- If movie IS present on Plex in 1080p, it'll just send the 4K request as intended by the user

**Comment by kiranshila** [comment]
I guess what I'm missing is that I don't understand the problem, mainly this bit

> Trying to request it in 1080p was not possible, and I had to manually add the item/movie through the default Radarr instance.

I'd rather not add something counterintuitive (even if behind an option) to fix an upstream bug. Like, I don't think `Request 4K` should do anything besides requesting 4K. My proposal of another button was to make it very obvious what it is doing.

What exactly is the bug? Have you opened an issue with Overseerr?

**Comment by VP-EN** [comment]
No, I haven't opened an issue with Overseerr. Doing all the reproducing steps is kind of a hassle, especially since reproducing this bug would take days in my case (adding different 4K series to Plex, removing them, testing, etc). So if I can avoid it, I'll prefer that 😄 .. Especially if I can't reproduce the bug, then I'm pretty sure that bug report would lead to nowhere.

I think I've experienced it once, and just made a mental note of it. I added the series Amazing Stories through my 4K Sonarr instance, and I thought I already had a 1080p version of it on Plex. 

Days later, when the 4K Amazing Stories was on my Plex, I checked my regular 1080p TV Series library on Plex, realizing it wasn't in there, and now only available in 4K. No biggie I thought, and tried to request the series in Overseerr. But Overseerr had registered the series as being "Available" and "Available in 4K", even though it was only available in 4K on Plex - which then means I couldn't request it through Overseerr. I tried rescanning the Plex library in Overseerr, but that didn't help.

I ended up just adding the series through the regular Sonarr instance.

Anyways... 😄 

> My proposal of another button was to make it very obvious what it is doing.
Yeah, I guess that would be the best option then.. I think the `Request Both` button-prompt could work, if the message was clear enough, like: "The movie you are requesting (in 4K) is not available. Please request it on both profiles", then showing the `Request Both` button... or something like that

**Comment by VP-EN** [comment]
@kiranshila Hey mate, is there any news from Doplarr land? 😄 . Are you still very busy with your PhD? 🙂

**Comment by kiranshila** [comment]
> @kiranshila Hey mate, is there any news from Doplarr land? 😄 . Are you still very busy with your PhD? 🙂

Hey! Yeah the term is ending soon, which will give me the time to finish up all the stuff I was trying to do.
